ml_service/pipelines/run_train_pipeline.py

```diff
@@ -1,24 +1,34 @@
 from azureml.pipeline.core import PublishedPipeline
 from azureml.core import Workspace
-from azureml.core.authentication import ServicePrincipalAuthentication
 import os
 import sys
+import argparse
 sys.path.append(os.path.abspath("./ml_service/util"))  # NOQA: E402
 from env_variables import Env
 
 
 def main():
+
+    parser = argparse.ArgumentParser("register")
+    parser.add_argument(
+        "--output_pipeline_id_file",
+        type=str,
+        help="Name of a file to write pipeline ID to"
```

Comment:
Would be nice to set a default file name so no need to pass a filename with the arg.

Reply:
Adding it as default, but still I prefer to make its value explicit in the call below, as I think it makes the logic of the pipeline clearer. Convention over configuration is good up to the point where it makes threads hard to follow :)

```
          python $(Build.SourcesDirectory)/ml_service/pipelines/run_train_pipeline.py --output_pipeline_id_file "pipeline_id.txt" --skip_train_execution
          # Set AMLPIPELINEID variable for next AML Pipeline task in next job
          AMLPIPELINEID="$(cat pipeline_id.txt)"
```